walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -487,9 +495,244 @@ public void noTimeWithOnlyOptionals() {
+                "Event 3",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0900AM, false),
+                Arrays.asList(PERSON_B)),
+            new Event(
+                "Event 4",
+                TimeRange.fromStartEnd(TIME_0800AM, TIME_1100AM, false),
+                Arrays.asList(PERSON_C)),
+            new Event(
+                "Event 5",
+                TimeRange.fromStartEnd(TIME_1200AM, TimeRange.END_OF_DAY, false),
+                Arrays.asList(PERSON_D)));
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_B);
+    request.addOptionalAttendee(PERSON_C);
+    request.addOptionalAttendee(PERSON_D);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TIME_1230AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noneOfOptionalsFreeTimesWorksWithAttendees() {
+
+    Collection<Event> events =
+        Arrays.asList(
+            new Event(
+                "Event 1",
+                TimeRange.fromStartEnd(TIME_0830AM, TIME_1000AM, false),
+                Arrays.asList(PERSON_A)),
+            new Event(
+                "Event 2",
+                TimeRange.fromStartEnd(TIME_1100AM, TIME_1330AM, false),
+                Arrays.asList(PERSON_A)),
+            new Event(
+                "Event 3",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_1100AM, false),
+                Arrays.asList(PERSON_B)),
+            new Event(
+                "Event 4",
+                TimeRange.fromStartEnd(TIME_1330AM, TimeRange.END_OF_DAY, false),
+                Arrays.asList(PERSON_B)),
+            new Event(
+                "Event 5",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TimeRange.END_OF_DAY, true),
+                Arrays.asList(PERSON_C)),
+            new Event(
+                "Event 6",
+                TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_1130AM, false),
+                Arrays.asList(PERSON_D)),
+            new Event(
+                "Event 7",
+                TimeRange.fromStartEnd(TIME_1330AM, TimeRange.END_OF_DAY, false),
+                Arrays.asList(PERSON_D)));
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_1_HOUR);
+    request.addOptionalAttendee(PERSON_B);
+    request.addOptionalAttendee(PERSON_C);
+    request.addOptionalAttendee(PERSON_D);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(
+            TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            TimeRange.fromStartEnd(TIME_1000AM, TIME_1100AM, false),
+            TimeRange.fromStartEnd(TIME_1330AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
```

Comment:
I noticed none of your tests use an event that has multiple attendees. It'd be good to modify at least one of the events in at least one of the tests to have more than 1 attendee.